### ISSUE TITLE
False-positive result on add nullable type in final method of abstract class

### ISSUE THREAD

**Issue by vjik** [Roave/BackwardCompatibilityCheck#313]
Example:

```php
abstract class X {
	final public function test(string $var): self
}
```

changed to:

```php
abstract class X {
	final public function test(?string $var): self
}
```

result will be:

```
[BC] CHANGED: The parameter $$ of X#test() changed from string to ?string
```

Real example: https://github.com/yiisoft/html/pull/78

**Comment by Ocramius** [comment]
So, if I understand correctly, if a method is `final`, then a contravariant type (expansion) is acceptable here.

The check should be this one:

https://github.com/Roave/BackwardCompatibilityCheck/blob/7f185e3128276bbaa3d6c086ebee45e3ba02c374/src/DetectChanges/BCBreak/FunctionBased/ParameterTypeChanged.php#L22

I think we need to restrict `ParameterTypeChanged` checks (see https://github.com/Roave/BackwardCompatibilityCheck/blob/7f185e3128276bbaa3d6c086ebee45e3ba02c374/bin/roave-backward-compatibility-check.php) to only non-contravariant, when the method is `final`.

**Comment by Ocramius** [comment]
Not going to be able to drag this into `6.0` - removing milestone for now.

**Comment by vjik** [comment]
Any news about this issue?

**Comment by Ocramius** [comment]
Nobody working on this: send a patch?

**Comment by vjik** [comment]
> Nobody working on this: send a patch?

Is `/bin/roave-backward-compatibility-check.php` logic not being tested?

**Comment by Ocramius** [comment]
No, it's effectively a default configuration composed.

Smoke-testing it may make sense, but fully testing it is combinatory logic / too much
